src/interceptors/fetch/index.ts

```diff
@@ -70,10 +70,17 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
         return mockedResponse
       })
 
-      if (resolverResult.error || requestAbortRejection.state === 'rejected') {
+      if (requestAbortRejection.state === 'rejected') {
```

Comment:
Do you think we should conceptually treat this promise as resolved instead? 

If we imagine promise as an intention, then the intention here is to know when the request has been aborted by another actor. In that light, an aborted request would mean a resolved `requestAborted` promise. 

It would make this check read much nicer as well. What are your thoughts on this? 

I'm also a bit hesitant to base this condition on the request abort promise itself. I'd rather we used that promise in the `Promise.race()` but handled the `resolverResult.error` differently. This implies that `requestAbortRejection` can still throw but it does so with a custom error/object that is later checked on the `resolverResult.error`:

```js
signal.addEventListener('abort', () => {
  requestAbortRejection.reject({ type: 'aborted', reason: signal.reason })
})

if (resolverResult.error) {
  if (resolverResult.error.type === 'aborted') {
    return Promise.reject(resolverResult.error.reason)
  }

  return /* create the TypeError, set "cause" */
}
```

Reply:
It is true that the promise encapsulate the expectation of an aborted request.
Resolving it would mean that the expectation has been reached, which is logically correct.
However, in our case, we use a promise as a poor man's event : the promise rejection carries no expectation and therefore no logical meaning. We just want to be notified when the abortion happens.
From there, this promise is a mere async tool.
Resolving it, then check the race winner, then throw a custom object, check the resolverResult resolution value, all of that to ultimately throw the error we had in the first place, seems unnecessary to me.
On a side note, in the end, I'm not the maintainer of the project, if you feel more comfortable with this extra layer, do it, this is your call :)

Reply:
I ended up leaving the explicit rejection promise check but instead of coupling the request rejection promise and the resolver error, I reject the request promise with the abort rejection reason. Also, that is hell of a sentence to write.